src/lib.rs

```diff
@@ -367,8 +369,13 @@ impl<T: Data> UiState<T> {
     fn do_event(&mut self, event: Event, win_ctx: &mut dyn WinCtx) -> bool {
         // should there be a root base state persisting in the ui state instead?
         let mut base_state = Default::default();
+        let mut cursor = match event {
+            Event::MouseMoved(..) => Some(Cursor::Arrow),
```

Comment:
This seems weird? If the mouse moved but we're dragging an item, for instance, we don't want arrow.

Reply:
Maybe I should add a comment to clarify, but the intent here is that arrow is the default. Basically the contract is that a widget that sets the cursor should *always* set it in its `MouseMoved` handler.

Reply:
Ah okay I misunderstood, but am still a little confused.

What is the distinction of `cursor` being `Some` or `None`, here? Afaict in either case we allow the widget to call `set_cursor`, and it also seems like this means that we will always be calling `set_cursor` on each `MouseMoved` event, which seems unnecessary?

Reply:
On Windows I'm pretty sure that is indeed what you want - it's hard to be sure if something else has set the cursor, so always setting it seems safe. Also I think the call is pretty cheap.

On mac if we end up setting tracking rectangles, then it's pretty clear that setting it to the same value accomplishes nothing, so I'd be fine with checking equality with the past value on the Rust side before calling into Cocoa.

Reply:
I think I'd feel better if we knew the last cursor we'd set, and only called the system api if necessary? At the least it might be saving us a context switch? This isn't very substantiated but it feels unnecessary.

It also makes me think that it might be worth doing this in something like `ActiveChanged`, instead of `MouseMoved`? still just spitballing.

Reply:
I think I'd feel better if we knew the last cursor we'd set, and only called the system api if necessary? At the least it might be saving us a context switch? This isn't very substantiated but it feels unnecessary.

It also makes me think that it might be worth doing this in something like `ActiveChanged`, instead of `MouseMoved`? still just spitballing.

Reply:
I think it'd be `HotChanged`, as that gets triggered on hover, but active state only changes on mouse press. I can make that change, but it will require a tad bit more event flow to get right, particularly (on Windows) I need to make sure that all hot state is reset when the mouse leaves the window. This turns out to be slightly tricky, I think I need to call [`TrackMouseEvent` ](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-trackmouseevent) to ensure that I get a WM_MOUSELEAVE. If I don't do this, then I'll skip setting the cursor when (eg) the mouse moves from a nonclient region such as a window sizing border back into the main window; the app will think the cursor is set, but in reality it was set in the hover over the nonclient region. Just doing it on mouse moved seems to be the windows way and avoids this extra complication. I can make this change if you feel strongly (an argument can be made that it gives more precise hot highlighting anyway).

Incidentally, https://devblogs.microsoft.com/oldnewthing/20031013-00/?p=42193 makes some interesting reading, explaining why there's WM_MOUSELEAVE but no WM_MOUSEENTER. The example is problematic, as it doesn't mention anything about TrackMouseEvent and I'm not seeing the WM_MOUSELEAVE events on my system.

I'm still confused about mac.